app/src/org/commcare/dalvik/activities/HomeActivityUIController.java

```diff
@@ -0,0 +1,391 @@
+        }
+
+        grid.getViewTreeObserver().addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
+            @SuppressLint("NewApi")
+            @Override
+            public void onGlobalLayout() {
+                if (Build.VERSION.SDK_INT < Build.VERSION_CODES.JELLY_BEAN) {
+                    grid.getViewTreeObserver().removeGlobalOnLayoutListener(this);
+                } else {
+                    grid.getViewTreeObserver().removeOnGlobalLayoutListener(this);
+                }
+                grid.requestLayout();
+                adapter.notifyDataSetChanged(); // is going to populate the grid with buttons from the adapter (hardcoded there)
+                configUI();
+            }
+        });
+    }
+
+    protected void configUI() {
+        if (CommCareApplication._().getCurrentApp() == null) {
+            // This method depends on there being a seated app, so don't proceed with it if we
+            // don't have one
+            return;
+        }
+        setupVersion();
+        setupButtons();
+        activity.rebuildMenus();
+    }
+
+    private void setupVersion() {
+        TextView version = (TextView) activity.findViewById(R.id.str_version);
+        if (version != null) {
+            version.setText(CommCareApplication._().getCurrentVersionString());
+        }
+    }
+
+    private void setupButtons() {
+        setupStartButton();
+        setupIncompleteFormsButton();
+        setupLogoutButton();
+        setupViewOldFormsButton();
+        setupSyncButton();
+    }
+
+    // region - setup methods for all buttons
+
+    private void setupStartButton() {
+        startButton = adapter.getButton(R.layout.home_start_button);
+        if (startButton != null) {
+            startButton.setText(Localization.get("home.start"));
+        }
+        adapter.setOnClickListenerForButton(R.layout.home_start_button, getStartButtonListener());
+    }
+
+    private void setupIncompleteFormsButton() {
+        viewIncompleteFormsButton = adapter.getButton(R.layout.home_incompleteforms_button);
+        if (viewIncompleteFormsButton != null) {
+            setIncompleteFormsText(CommCareApplication._().getSyncDisplayParameters());
+        }
+        adapter.setOnClickListenerForButton(R.layout.home_incompleteforms_button, getIncompleteButtonListener());
+    }
+
+    private void setupLogoutButton() {
+        logoutButton = adapter.getButton(R.layout.home_disconnect_button);
+        if (logoutButton != null) {
+            logoutButton.setText(Localization.get("home.logout"));
+            logoutButton.setNotificationText(activity.getActivityTitle());
+            adapter.notifyDataSetChanged();
+        }
+        adapter.setOnClickListenerForButton(R.layout.home_disconnect_button, getLogoutButtonListener());
+    }
+
+    private void setupViewOldFormsButton() {
+        SquareButtonWithNotification viewOldForms = adapter.getButton(R.layout.home_savedforms_button);
```

Comment:
Because it's the only button that isn't referenced by other methods. But I can change it for consistency purposes.